src/client.ts

```diff
@@ -643,19 +646,21 @@ export class Client extends EventEmitter {
     }
 
     if (this.hardReconnectRequired) {
-      let jidPromise: Promise<any>;
-      if (this.config.jid) {
```

Comment:
In the previous code, we would automatically resolve our promise if a JID was passed in via config. What's weird though is we also grabbed the value from the promise and assigned it to `config.jid` even though that value is the same if we end up resolving without needing to fetch via api.

Reply:
Huh...